tests/fs/test_getcwd_with_non_ascii_name.cpp

```diff
@@ -0,0 +1,19 @@
+#include <cassert>
+#include <fstream>
+#include <sys/stat.h>
+#include <unistd.h>
+
+int main()
+{
+    const auto* const dirName = u8"абвг";
+
+    auto ret = mkdir(dirName, 0777);
+    assert(ret == 0);
+    ret = chdir(dirName);
+    assert(ret == 0);
+
+    char buf[255];
+    auto ret2 = getcwd(buf, sizeof(buf));
```

Comment:
Could you update this test to try out a longer path name, and also test the negative case of truncation if the path exceeds `sizeof(buf)`, and verify that this does not overflow the destination space?